### ISSUE TITLE
Usage in Rails 6

### ISSUE THREAD

**Issue by evanliewer** [GetJobber/recurring_select#138]
Has anyone successfully been able to install recurring_select in Rails 6? I get no errors on my form, but clicking on the dropdown does nothing.  I tried adding the resources through asset pipeline, but have not had luck with it so far.  Any success stories?

**Comment by rossinkiwi** [comment]
I've had success with this fork:
https://github.com/Syndicode/recurring_select/tree/rails_6_support

I'm using sprockets which means I had to do some work to get jQuery working:
````
#app/views/layouts/application.html.erb
  <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
  <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
  <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
````
````
#app/javascript/packs/application.js
import $ from "jquery"
import jQuery from "jquery"
````
````
#config/webpack/environment.js
const { environment } = require('@rails/webpacker')

const webpack = require('webpack');
environment.plugins.append('Provide', new webpack.ProvidePlugin({
  $: 'jquery/src/jquery',
  jQuery: 'jquery/src/jquery',
}));

module.exports = environment
````

````
# app/assets/javascripts/application.js
//= require recurring_select
````
````
# app/assets/stylesheets/application.scss
*= require recurring_select
````
Also had to add the following callback to my model (see https://github.com/GetJobber/recurring_select/issues/13):
````
before_save { |record| record.recurrence_rule = nil if record.recurrence_rule == 'null' }
````

Hope this helps.  Upgrading to rails6 was non-trivial for me, hopefully your ride will be smoother!

**Comment by evanliewer** [comment]
Thank you so much!!!! This was so helpful and was able to get it working for me.  I was also following the GoRails tutorial for recurring select, and I had two changes needed to get it to work in Rails 6. 

First I got a weird error from the application.scss file.  I skipped that and instead added "@import 'recurring_select';" to my already existing scss file.

The second I dont fully grasp why it isn't working, as it used to. But I had to add the && value != "null" to the model to make it work.  Otherwise it was processing null files as a rule.  Adding the && to the code below has been working.

 def recurring=(value)
    if RecurringSelect.is_valid_rule?(value) && value != "null"
      super(RecurringSelect.dirty_hash_to_rule(value).to_hash)
    else
      super(nil)
    end
  end

**Comment by garethfr** [comment]
I'm using rails6 with webpacker and I had some problems getting things up and running due to conflicts between required coffee-rails package and webpacker coffeescript package . Here's what I did to fix things:

- remove coffee-rails dependency in recurring_select.gemspec
- add following in package.json
`
   "@rails/webpacker": "4.3.0",
     "coffee-loader": "^1.0.1",
     "coffeescript": "^2.5.1",
`
(added with `rails webpacker:install:erb` and `rails webpacker:install:coffee`)

- added `.coffee.erb` to `config/webpacker.yml`  under `extensions`

A fork of the original repo with these changes is at https://github.com/inkommon/recurring_select

**Comment by jessevanherk** [comment]
this was addressed with the 3.0.0 release.
